### PR TITLE
Add iterator support to brute force search

### PR DESCRIPTION
**Pull request by zhengbuqian**
issue: #367 
added unit test for fp32 type

https://github.com/milvus-io/milvus/issues/30307

/kind enhancement

### CODE REVIEW

File: src/common/comp/brute_force.cc [hunk 1]
```diff
@@ -438,6 +439,142 @@ BruteForce::SearchSparse(const DataSetPtr base_dataset, const DataSetPtr query_d
...
+ private:
+    // sort the next sort_size_ elements
+    inline void
+    sort_next() {
+        if (next_ < sorted_) {
+            return;
+        }
+        size_t current_end = std::min(results_.size(), sorted_ + sort_size_);
+        std::nth_element(results_.begin() + sorted_, results_.begin() + current_end, results_.end(), comp_);
+        std::sort(results_.begin() + sorted_, results_.begin() + current_end, comp_);
```
Comment (alexanderguzhva) [inline src/common/comp/brute_force.cc:471]: `std::partial_sort()`

File: src/common/comp/brute_force.cc [hunk 2]
```diff
@@ -438,6 +439,142 @@ BruteForce::SearchSparse(const DataSetPtr base_dataset, const DataSetPtr query_d
...
+                return true;
+            }
+            return larger_is_closer ? a.first > b.first : a.first < b.first;
+        }
+    } comp_;
+
+    std::vector<std::pair<float, int64_t>> results_;
+    size_t next_ = 0;
+    size_t sorted_ = 0;
+    size_t sort_size_;
```
Comment (alexanderguzhva) [inline src/common/comp/brute_force.cc:494]: `size_t sort_size_ = 0;`

**Review by alexanderguzhva — COMMENTED** [review]
nits

**Comment by alexanderguzhva** [comment]
/lgtm
overall

File: src/common/comp/brute_force.cc (hunk 1 above)
Comment (zhengbuqian) [reply src/common/comp/brute_force.cc]: done

**Comment by zhengbuqian** [comment]
issue: https://github.com/zilliztech/knowhere/issues/367

both this and IVFFLAT iterators are for growing segment, using that as the umbrella issue

**Pull request merged**